nebari/upgrade.py

```diff
@@ -385,6 +386,32 @@ def _version_specific_upgrade(
         return config
 
 
+class Upgrade_2023_4_2(UpgradeStep):
+    version = "2023.4.2"
+
+    def _version_specific_upgrade(
+        self, config, start_version, config_filename: pathlib.Path, *args, **kwargs
+    ):
+        """
+        Prompt users to delete Argo CRDs
+        """
+
+        kubectl_delete_argo_crds_cmd = "kubectl delete crds clusterworkflowtemplates.argoproj.io cronworkflows.argoproj.io workfloweventbindings.argoproj.io workflows.argoproj.io workflowtasksets.argoproj.io workflowtemplates.argoproj.io"
+
+        rich.print(
+            f"\n\n[bold cyan]Note:[/] Upgrading requires a one-time manual deletion of the Argo Workflows Custom Resource Definitions (CRDs). \n\n[red bold]Warning:  [link=https://{config['domain']}/argo/workflows]Workflows[/link] and [link=https://{config['domain']}/argo/workflows]CronWorkflows[/link] created before deleting the CRDs will be erased when the CRDs are deleted and will not be restored.[/red bold] \n\nThe updated CRDs will be installed during the next [cyan bold]nebari deploy[/cyan bold] step. Argo Workflows will not function after deleting the CRDs until the updated CRDs are installed in the next nebari deploy. You must delete the Argo CRDs before upgrading to {self.version} or deploy step will fail.  Please delete them before proceeding by generating a kubeconfig (see [link=https://www.nebari.dev/docs/how-tos/debug-nebari/#generating-the-kubeconfig]docs[/link]), installing kubectl (see [link=https://www.nebari.dev/docs/how-tos/debug-nebari#installing-kubectl]docs[/link]), and running the following command:\n\n\t[cyan bold]{kubectl_delete_argo_crds_cmd} [/cyan bold].\n"
+            ""
+        )
+
+        continue_ = Prompt.ask("Have you deleted the Argo CRDs? \[y/N]", default="N")
+        if not continue_ == "y":
+            raise ValueError(
```

Comment:
Can we just `exit()` instead?

```python
if not continue_ == "y":
    print(f"You must delete the Argo CRDs before upgrading to {self.version}")
    exit()
```

This avoids the large traceback being printed to the screen and allows the user to see the command they need to run.

Reply:
good idea, I pushed a commit